### PR TITLE
Adding Shopify GraphQL Tool Spec and Demo

### PR DESCRIPTION
**Pull request by ajhofmann**

### CODE REVIEW

File: llama_hub/tools/shopify/base.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+"""Shopify tool spec."""
+
+from llama_index.tools.tool_spec.base import BaseToolSpec
+from typing import Optional
+import shopify
+
+class ShopifyToolSpec(BaseToolSpec):
```
Comment (jerryjliu) [inline llama_hub/tools/shopify/base.py:7]: out of curiosity could we create a general "GraphQLAPIToolSpec" that could run graphql queries against any graphql http endpoint? 

to be clear we should def keep this one

**Review by jerryjliu — COMMENTED** [review]
this demo is great

File: llama_hub/tools/shopify/base.py (hunk 1 above)
Comment (ajhofmann) [reply llama_hub/tools/shopify/base.py:7]: Will work on splitting out a GraphQLAPIToolSpec and SchemaDefinitionLanguageLoader today 👌

File: llama_hub/tools/shopify/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+In particular, the tool is very effective when combined with a method of retriving data from the GraphQL schema defintion.
+
+```python
+from llama_hub.tools.shopify.base import ShopifyToolSpec
+from llama_index.agent import OpenAIAgent
+
+tool_spec = ShopifyToolSpec('your-store.myshopify.com', '2023-04', 'your-api-key')
+
+agent = OpenAIAgent.from_tools(tool_spec.to_tool_list())
```
[the remark this answers is not in the document]
Comment (ajhofmann) [reply llama_hub/tools/shopify/README.md]: Updated!

**Review by jerryjliu — APPROVED** [review]

**Pull request merged**